app/models/energy/constructionset.py

```diff
@@ -0,0 +1,146 @@
+"""Construction Set Schema"""
+from pydantic import BaseModel, Schema, validator, ValidationError
+from typing import List, Union
+from enum import Enum
+from uuid import UUID, uuid4
+from datetime import datetime
+from app.models.energy.construction import EnergyConstructionOpaque, EnergyConstructionWindow
+
+
+class WallSet(BaseModel):
+    """A set of constructions for wall assemblies."""
+
+    type: Enum('WallSet', {
+               'type': 'WallSet'})
+
+
+    interior_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    ground_wall: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+
+class FloorSet(BaseModel):
+    """A set of constructions for floor assemblies."""
+
+    type: Enum('FloorSet', {
+               'type': 'FloorSet'})
+
+
+    interior_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    ground_floor: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+class RoofCeilingSet(BaseModel): 
+    """A set of constructions for roof and ceiling assemblies."""
+
+    type: Enum('RoofCeilingSet', {
+               'type': 'RoofCeilingSet'})
+
+    interior_ceiling: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+    exterior_roof: EnergyConstructionOpaque = Schema(
+        default=None
+    )    
+
+    ground_ceiling: EnergyConstructionOpaque = Schema(
+        default=None
+    )
+
+class ApertureSet(BaseModel): 
+    """A set of constructions for aperture assemblies."""
+
+    type: Enum('ApertureSet', {
+               'type': 'ApertureSet'})
+
+    interior_window: EnergyConstructionWindow = Schema(
```

Comment:
The one exception to the comment that I made about the properties on the `ApertureSet` is that I think this one should just be called `interior_construction`.  This is because I am imagining that this construction gets applied to all interior apertures, whether they are fixed, operable, a glass door or in an interior ceiling.  It's pretty rare that someone wants to use a construction for interior glazed surfaces that is not just a single pane of glass (I feel that this is true whether it's for an interior glass door or an interior window).  So, for the `ConstructionSet`, I think we should just support one type of interior aperture construction.  If people really need to customize different interior constructions, they can do so surface-by-surface.